website/server/controllers/api-v3/auth.js

```diff
@@ -643,6 +643,47 @@ api.updateEmail = {
 };
 
 /**
+ * @api {post} /api/v3/user/auth/reset-api-token Reset API Token
+ * @apiDescription Reset the user's API Token
+ * @apiName ResetAPIToken
+ * @apiGroup User
+ *
+ * @apiParam (Body) {String} password the user password or current API token
+ *
+ * @apiSuccess {String} data.apiToken The new API token
+ */
+api.resetAPIToken = {
+  method: 'POST',
+  middlewares: [authWithHeaders()],
+  url: '/user/auth/reset-api-token',
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    let authType = 'local';
+    if ((user.auth.facebook.id || user.auth.google.id) && !user.auth.local.email)  {
+      authType = 'social';
+    }
+
+    req.checkBody('password', res.t('invalidReqParams')).notEmpty();
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let password = req.body.password;
+    if (authType === 'local') {
+      let isValidPassword = await passwordUtils.compare(user, password);
+      if (!isValidPassword) throw new NotAuthorized(res.t('wrongPassword'));
+    } else if (authType === 'social') {
+      if (password !== res.t('reset')) throw new NotAuthorized(res.t('incorrectResetPhrase'));
```

Comment:
the reset text seems to be localized (res.t) here but not in the client side like here https://github.com/HabitRPG/habitica/pull/9320/files#diff-b90f9bd8f348e5c8e16dbb557ac44db9R22